specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/azureWebCategory.json

```diff
@@ -0,0 +1,184 @@
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "$expand",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "Expands resourceIds back referenced by the azureWebCategory resource."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns the resulting azureWebCategory resource.",
+            "schema": {
+              "$ref": "#/definitions/AzureWebCategory"
+            }
+          },
+          "default": {
+            "description": "Unexpected error.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get Azure Web Category by name": {
+            "$ref": "./examples/AzureWebCategoryGet.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Network/azureWebCategories": {
+      "get": {
+        "tags": [
+          "AzureWebCategories"
+        ],
+        "operationId": "AzureWebCategories_ListBySubscription",
+        "description": "Gets all the Azure Web Categories in a subscription.",
+        "parameters": [
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success. The operation returns a list of Azure Web Category resources.",
+            "schema": {
+              "$ref": "#/definitions/AzureWebCategoryListResult"
+            }
+          },
+          "default": {
+            "description": "Unexpected Error.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List all Azure Web Categories for a given subscription": {
+            "$ref": "./examples/AzureWebCategoriesListBySubscription.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "AzureWebCategoryPropertiesFormat": {
+      "properties": {
+        "group": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The name of the group that the category belongs to."
+        }
+      },
+      "description": "Azure Web Category Properties."
+    },
+    "AzureWebCategory": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/AzureWebCategoryPropertiesFormat",
+          "description": "Properties of the Azure Web Category."
+        },
+        "id": {
```

Comment:
rather than having the resource properties inline, you should use allOf to bring in a pre-defined resource type, like this: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/network/resource-manager/Microsoft.Network/stable/2020-06-01/firewallPolicy.json#L571

Note that there are definitions in network for both resource and subResource, so you should be able to use one.

Reply:
"allOf": [
        {
          "$ref": "./network.json#/definitions/Resource"
        } has location properties. This resource is a global resource.  Please see the commit https://github.com/Azure/azure-rest-api-specs/pull/10918/commits/b5e1f0980a07af3eb0e386bbfc41a69110856d6c on master PR

Reply:
Then it should use proxy resource.  There should be some indication that this is a resource